packages/core/src/Account.ts

```diff
@@ -245,43 +250,67 @@ export class Account<T = any> extends EventEmitter {
    */
   public async init(
     clientType: ClientType,
-    {cookie, initClient = true, onNewClient}: InitOptions = {},
+    {cookie, initClient = true, onNewClient, entropyData}: InitOptions = {},
   ): Promise<Context> {
     const context = await this.apiClient.init(clientType, cookie);
     await this.initServices(context);
 
-    /** @fixme
-     * When we will start migrating to CoreCrypto encryption/decryption, those hooks won't be available anymore
-     * We will need to implement
-     *   - the mechanism to handle messages from an unknown sender
-     *   - the mechanism to generate new prekeys when we reach a certain threshold of prekeys
-     */
-    this.service!.cryptography.setCryptoboxHooks({
-      onNewPrekeys: async prekeys => {
-        this.logger.debug(`Received '${prekeys.length}' new PreKeys.`);
+    if (initClient && this.cryptoProtocolConfig?.proteus) {
+      const coreCryptoKeyId = 'corecrypto-key-proteus';
+      const dbName = `${this.generateSecretsDbName(context)}-proteus`;
+      const secretStore = this.cryptoProtocolConfig?.systemCrypto
+        ? await createCustomEncryptedStore(dbName, this.cryptoProtocolConfig?.systemCrypto)
+        : await createEncryptedStore(dbName);
 
-        await this.apiClient.api.client.putClient(context.clientId!, {prekeys});
-        this.logger.debug(`Successfully uploaded '${prekeys.length}' PreKeys.`);
-      },
+      let key = await secretStore.getsecretValue(coreCryptoKeyId);
 
-      onNewSession: onNewClient,
-    });
+      if (!key) {
```

Comment:
Why do we call the same function twice if it fails to return a key to us? This code is hard to understand for me.